src/main/java/Model/FEN.java

```diff
@@ -0,0 +1,187 @@
+package Model;
+
+
+import Model.pieces.Piece;
+
+import static java.util.Objects.isNull;
+
+public class FEN {
+
+    private String[] fenPosition = new String[8];
+    private String[] fenBreaks = {"/","/","/","/","/","/","/"," ",};
+    private Boolean whitesTurn = false;
+    private Boolean[] whiteCastling = {true, true};
+    private Boolean[] blackCastling = {true, true};
+    private Piece enPassantPiece = null;
+    private int halfmove = 0;
+    private int fullmove = 0;
+    private String output;
+
+    public FEN(Piece[][] piece2DArray){
+        for (int rank = 0; rank < 8; rank++){
+                updateFENPosition(piece2DArray[rank],rank);
+            }
+    }
+
+    //with all the pieces in the rank, update the appropriate FEN rank in FENPosition
+    public void updateFENPosition(Piece[] pieceRank, int rank){
+        String output = "";
+        int emptySquare = 0;
+        for (Piece piece:pieceRank) {
+            if (isNull(piece)) {
+                emptySquare += 1;
+            } else {
+                if (!(emptySquare == 0)) {
+                    output += emptySquare;
+                    emptySquare = 0;
+                }
+                output += getFENletter(piece);
+
+            }
+        }
+        if (!(emptySquare==0)) {
+            output += emptySquare;
+        }
+        fenPosition[rank]=output;
+    }
+
+    //From the Piece arg, get the correct FEN letter
+    private String getFENletter(Piece piece){
+        String output = "";
+        switch (piece.getPieceType()){
+            case King:
+                output+="k";
+                break;
+            case Queen:
+                output+="q";
+                break;
+            case Rook:
+                output+="r";
+                break;
+            case Knight:
+                output+="n";
+                break;
+            case Bishop:
+                output+="b";
+                break;
+            case Pawn:
+                output+="p";
+                break;
+        }
+        if(piece.getIsWhite()){
+            output = output.toUpperCase();
+        }
+        return output;
+    }
```

Comment:
why not encapsulate this information within the `PieceType` class? can do something like:
```java
public enum PieceType {
    King("k"),
    Queen("q"),
     ...;
    
    public final String fenRepresentation;

    private PieceType(String fenRepresentation) {
        this.fenRepresentation = fenRepresentation;
    }

    public String getFenRepresentation(boolean isWhite) {
        return isWhite ? this.fenRepresentation.toUpperCase() : this.fenRepresentation;
    }
}
```

Reply:
Implemented. Didn't even know this was possible!!